scapy/contrib/bgp.py

```diff
@@ -1,169 +1,676 @@
-#! /usr/bin/env python 
-
-# http://trac.secdev.org/scapy/ticket/162
-
+#! /usr/bin/env python
+"""BGP-4 protocol with mulit-protocol support"""
 # scapy.contrib.description = BGP
 # scapy.contrib.status = loads
 
-from scapy.packet import *
+from scapy.packet import Packet, Raw, bind_layers
 from scapy.fields import *
 from scapy.layers.inet import TCP
+from scapy.contrib.bgp_fields import *
 
+def _class_dispatcher(s, class_dict, def_class, index_from=None):
```

Comment:
Why is `index_from` defaulting to `None`, while it seems to have to be a function?